src/visual.ts

```diff
@@ -260,122 +273,163 @@ export class Visual implements IVisual {
+        const maxGapWidth = Math.max(...gapWidths);
+        const firstValueMeasureIndex: number = 0, firstGroupIndex: number = 0, secondGroupIndex = 1;
+        const y_group0Values = grouped[firstGroupIndex]
             && grouped[firstGroupIndex].values[firstValueMeasureIndex]
             && grouped[firstGroupIndex].values[firstValueMeasureIndex].values;
-        let y_group1Values = grouped[secondGroupIndex]
+        const y_group1Values = grouped[secondGroupIndex]
             && grouped[secondGroupIndex].values[firstValueMeasureIndex]
             && grouped[secondGroupIndex].values[firstValueMeasureIndex].values;
 
-        let series: Series[] = [];
-        let dataPoints: DataPoint[] = [];
 
-        for (let categoryIndex = 0, seriesCategoryIndex = 0, len = timeStampColumn.values.length; categoryIndex < len; categoryIndex++ , seriesCategoryIndex++) {
+        let minSize: number = settings.dots.minSize.value;
+        let maxSize: number = settings.dots.maxSize.value;
+
+        const eventSizeScale: LinearScale = <LinearScale>Visual.createScale(
+            true,
+            columns.EventSize ? [d3Min(<number[]>columns.EventSize.values), d3Max(<number[]>columns.EventSize.values)] : [0, 0],
+            minSize,
+            maxSize);
+
+        const series: Series[] = [];
+        const dataPoints: DataPoint[] = Visual.generateDataPoints(timeStampColumn, categoryValues, host, columns, maxCategoryValue, minCategoryValue, settings, gapWidths, maxGapWidth, series, grouped, firstGroupIndex, dataPointLabelSettings, isScalar, y_group0Values, y_group1Values, eventSizeScale, runnerCounterFormatString, hasHighlights, valuesColumn);
+
+        behavior.applySelectionStateToData(dataPoints);
+
+        if (dataPoints.length > 0) {
+            series.push({
+                displayName: <string>grouped[firstGroupIndex].name,
+                lineIndex: series.length,
+                color: settings.series.fill.value.value,
+                data: dataPoints,
+                labelSettings: dataPointLabelSettings,
+                width: settings.series.width.value,
+                widthOfGap: 0
+            });
+        }
+        return series;
+    }
+
+    private static generateDataPoints(timeStampColumn: powerbiVisualsApi.DataViewCategoryColumn, categoryValues: any[], host: IVisualHost, columns: DataRoles<powerbiVisualsApi.DataViewCategoryColumn | powerbiVisualsApi.DataViewValueColumn>, maxCategoryValue: any, minCategoryValue: any, settings: PulseChartSettingsModel, gapWidths: number[], maxGapWidth: any, series: Series[], grouped: powerbiVisualsApi.DataViewValueColumnGroup[], firstGroupIndex: number, dataPointLabelSettings: ChartDataLabelsSettings, isScalar: boolean, y_group0Values: powerbiVisualsApi.PrimitiveValue[], y_group1Values: powerbiVisualsApi.PrimitiveValue[], eventSizeScale: LinearScale, runnerCounterFormatString: string, hasHighlights: boolean, valuesColumn: powerbiVisualsApi.DataViewValueColumn) {
```

Comment:
Please check all methods using so many arguments. If there are more than 3-4 arguments, create an object for this.

Reply:
Done.